rpcclient/infrastructure.go

```diff
@@ -1304,20 +1345,19 @@ func (c *Client) Connect(ctx context.Context, retry bool) error {
 	if c.config.HTTPPostMode {
 		return ErrNotWebsocketClient
 	}
-	if c.wsConn != nil {
+	if c.balancer.isReady {
```

Comment:
This is only defined on `RoundRobinBalancer`, so as mentioned elsewhere, this breaks the notion of the code being agnostic to the type of balancer being used.